Firebase/InAppMessaging/Flows/FIRIAMDisplayExecutor.m

```diff
@@ -307,14 +309,33 @@ - (void)displayMessageLoadError:(NSError *)error {
 
   UIAlertAction *defaultAction = [UIAlertAction actionWithTitle:@"OK"
                                                           style:UIAlertActionStyleDefault
-                                                        handler:^(UIAlertAction *action){
+                                                        handler:^(UIAlertAction *action) {
+                                                          self.alertWindow.hidden = NO;
+                                                          self.alertWindow = nil;
                                                         }];
 
   [alert addAction:defaultAction];
 
-  [[UIApplication sharedApplication].keyWindow.rootViewController presentViewController:alert
-                                                                               animated:YES
-                                                                             completion:nil];
+  dispatch_async(dispatch_get_main_queue(), ^{
+#if defined(__IPHONE_13_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
+    if (@available(iOS 13.0, *)) {
+      UIWindowScene *foregroundedScene = nil;
+      for (UIWindowScene *connectedScene in [UIApplication sharedApplication].connectedScenes) {
+        if (connectedScene.activationState == UISceneActivationStateForegroundActive) {
+          foregroundedScene = connectedScene;
+          break;
+        }
+      }
+      self.alertWindow = [[UIWindow alloc] initWithWindowScene:foregroundedScene];
+    }
+#else  // defined(__IPHONE_13_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 130000
+    self.alertWindow = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
+#endif
+    UIViewController *alertViewController = [[UIViewController alloc] init];
+    self.alertWindow.rootViewController = alertViewController;
```

Comment:
Is there a reason we're now adding a blank `UIViewController` to the root and presenting from it instead of presenting on the window itself?

Reply:
I tried setting the alert controller as the root but it was trying to stretch the alert to fit the whole screen.